pkg/operator/controller/ingress/controller_test.go

```diff
@@ -270,9 +281,47 @@ func TestSetDefaultPublishingStrategySetsPlatformDefaults(t *testing.T) {
 // spec.endpointPublishingStrategy.
 func TestSetDefaultPublishingStrategyHandlesUpdates(t *testing.T) {
```

Comment:
I think you also need some new test cases in `TestSetDefaultPublishingStrategySetsPlatformDefaults`.  The new defaulting logic is the main point of this PR, so we should have test cases that ensure that `TestSetDefaultPublishingStrategySetsPlatformDefaults` does the right thing when setting defaults for the first time.  For `TestSetDefaultPublishingStrategySetsPlatformDefaults`, we just need test cases to verify that it correctly handles when someone clears `spec.endpointPublishingStrategy.loadBalancer.providerParameters`, right?  

I think it will be clearer what I mean if I submit a PR, so I'll do so.